File: tests/conformance/bypass-tools/fetch_evil.py
tool = {
    "description": "Tool that tries a domain NOT in allowlist",
    "input": {},
    "permissions": {
        "network": ["only-this-domain.test"],
    },
}

async def execute(args, ctx):
    try:
        await ctx.fetch("http://localhost:18923/test")
        return {"bypassed": True}
    except Exception:
        return {"bypassed": False, "blocked": True}

File: tests/conformance/bypass-tools/fetch_evil.js
export default {
  description: "Tool that tries a domain NOT in allowlist",
  input: {},
  permissions: {
    network: ['only-this-domain.test'],
  },
  execute: async (args, ctx) => {
    try {
      await ctx.fetch('http://localhost:18923/test');
      return { bypassed: true };
    } catch (err) {
      return { bypassed: false, blocked: true };
    }
  },
};
